### PR TITLE
Add mail filter services

### PR DESCRIPTION
**Pull request by DavidePrincipi**
Add Rspamd and ClamAV containers for basic mail filtering.

- See container READMEs for details
- No UI and APIs are implemented

### CODE REVIEW

File: rspamd/etc/redis-volatile.conf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#
+# Redis volatile storage for Rspamd
```
[the remark this answers is not in the document]
Comment (gsanchietti) [reply rspamd/etc/redis-volatile.conf:2]: Do we use fuzzy worker on NS7?

[the remark this answers is not in the document]
Comment (DavidePrincipi) [reply rspamd/etc/redis-volatile.conf:2]: AFAIK yes

[the remark this answers is not in the document]
Comment (gsanchietti) [reply rspamd/etc/redis-volatile.conf:2]: I still prefer one redis instance, but the overhead should be really small.
So if it works, we can keep both instances.

File: dovecot/usr/local/lib/templates/local.conf [hunk 2]
```diff
@@ -108,7 +116,24 @@ service stats {
 # 90-sieve
 #
 plugin {
-  sieve_plugins = # sieve_extprograms sieve_imapsieve
+  sieve_plugins = sieve_extprograms sieve_imapsieve
+  sieve_global_extensions = +vnd.dovecot.pipe +vnd.dovecot.environment +editheader
+  sieve_editheader_max_header_size = 2048
+  sieve_before = /etc/dovecot/spam-actions.sieve
```
Comment (filippocarletti) [inline dovecot/usr/local/lib/templates/local.conf:122]:
```suggestion
  sieve_after = /etc/dovecot/spam-actions.sieve
```
Allow user filters to override junk filing

File: rspamd/etc/rspamd/local.d/antivirus.conf [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    scan_mime_parts = false;
+
+    # symbol to add (add it to metric if you want non-zero weight)
+    symbol = "CLAM_VIRUS";
+
+    type = "clamav";
+
+    # You can enable logging for clean messages
+    log_clean = true;
```
Comment (stephdl) [inline rspamd/etc/rspamd/local.d/antivirus.conf:18]: do not you fear to over charge the log with the logging of clean message. I think it is not enabled in NS7

**Review by stephdl — APPROVED** [review]
Good job, I suppose that we need to list and add other plugins ?

**Comment by DavidePrincipi** [comment]
> we need to list and add other plugins?

Not in this PR. I will add missing features (e.g. bypass rules) in another PR.

I will probably disable the fuzzy storage worker because we don't use it in ns7.

File: rspamd/etc/rspamd/local.d/antivirus.conf (hunk 3 above)
Comment (DavidePrincipi) [reply rspamd/etc/rspamd/local.d/antivirus.conf:18]: No, I don't. The Email app writes so many log lines for each _connection_! This is just one more...

The point is that Rspamd caches ClamAV response. I think that is an important information to better understand what is happening. Did Rspamd use a cached response? Did ClamAV analyze that message? Did it reach time-out?

**Comment by DavidePrincipi** [comment]
I will probably disable the fuzzy storage worker because we don't use it in ns7.

Added card to discuss again this point

https://trello.com/c/SrwRzbWd/221-mail-fuzzy-module

**Pull request merged**